### PR TITLE
[Task 6] Redux

### PR DESCRIPTION
**Pull request by zakhar98**
- All data fetches done as actions and received from store by components
- Creating, editing, deleting and updating (CRUD operations) of films are done as redux actions.
- Filtering by genre done as redux actions.
- Sorting by release date and rating is done as redux actions

### CODE REVIEW

File: src/services/api_service.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  static async createMovie(movie) {
+    await API.post(`/movies`, movie);
+  }
+
+  static async fetchMovie(movieId) {
+    const result = await API.get(`/movies/${movieId}`);
+    return result.data;
+  }
+}
```
Comment (IgorOmelyaniuk) [inline src/services/api_service.js:32]: Nice

File: src/redux/reducers/movies-list_reducer.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+       return {
+         ...state,
+         movies: {
+          items: action.payload.data,
+          totalCount: action.payload.totalAmount,
+         },
+         fetching: false,
+       };
+    case actionTypes.FETCH_MOVIES_FAILURE:
+      return {...state, fetching: false};
```
Comment (IgorOmelyaniuk) [inline src/redux/reducers/movies-list_reducer.js:37]: You could also add 'error' field to state for failure cases

File: src/components/SortPanel/SortPanel.js [hunk 3]
```diff
@@ -67,3 +65,26 @@ export default function SortPanel() {
...
+  }
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    fetchMovies: (params) => {dispatch(fetchMovies(params))},
+  }
+}
+
+export default connect(mapStateToProps, mapDispatchToProps)(SortPanel);
```
Comment (IgorOmelyaniuk) [inline src/components/SortPanel/SortPanel.js]: You can use short syntax without mapDispatchToProps, like this:
`export default connect(mapStateToProps, { fetchMovies })(SortPanel);`

**Review by IgorOmelyaniuk — APPROVED** [review]

File: src/redux/reducers/movies-list_reducer.js (hunk 2 above)
Comment (zakhar98) [reply src/redux/reducers/movies-list_reducer.js:37]: It makes sense if we'll have a component to display error details

File: src/components/SortPanel/SortPanel.js (hunk 3 above)
Comment (zakhar98) [reply src/components/SortPanel/SortPanel.js]: done

**Pull request merged**